src/rard/utils/duplication.py

```diff
@@ -0,0 +1,96 @@
+from bs4 import BeautifulSoup
+
+from rard.research.models import (
+    ApparatusCriticusItem,
+    Concordance,
+    OriginalText,
+    Translation,
+)
+
+
+def duplicate_original_text(original):
+    # Create a dictionary to hold the values for the new OriginalText object
+    new_original_text_data = {}
+
+    # Iterate over the fields of the OriginalText model
+    for field in original._meta.fields:
+        # Exclude the ID field
+        if field.name in ["id", "created", "modified"]:
+            continue
+
+        field_value = getattr(original, field.name)
+
+        new_original_text_data[field.name] = field_value
+
+    # Create a new OT object with the copied values
+    new_original_text = OriginalText.objects.create(**new_original_text_data)
+    return new_original_text
+
+
+def copy_concordances_apcrit_and_translations(original, new_original_text):
+    # Copy relationships to Concordances, Ap Crit & Translations
+    model_details = {
+        Concordance: {
+            "filter_name": "original_text",
+            "filter_value": original,
+            "ot_fieldname": "original_text",
+        },
+        ApparatusCriticusItem: {
+            "filter_name": "original_text",
+            "filter_value": original.pk,
+            "ot_fieldname": "parent",
+        },
+        Translation: {
+            "filter_name": "original_text",
+            "filter_value": original,
+            "ot_fieldname": "original_text",
+        },
+    }
+
+    for model_class in model_details.keys():
+        model_info = model_details[model_class]
+        filter_name = model_info["filter_name"]
+        filter_value = model_info["filter_value"]
+        ot_fieldname = model_info["ot_fieldname"]
```

Comment:
Could be written as:
```suggestion
    for model_class, model_info in model_details.items():
        filter_name = model_info["filter_name"]
        filter_value = model_info["filter_value"]
        ot_fieldname = model_info["ot_fieldname"]
```